### PR TITLE
feat: add support for more papyrus functions and enchance gamemode functionality

### LINKED ISSUE #953: Implement pies trading
See document
https://docs.google.com/document/d/1WUDEdZYHF1cux3rQkfpk26MO0VC9pLltFmxJHWQ0U94/edit?usp=drivesdk

### LINKED ISSUE #997: [Bug]: The quantity of "Gold001" (0000000F) is not displayed correctly in the inventory.
### Severity

Major. A crucial error that indicates a deviation from business logic or disrupts the program, but doesn't have a vital impact on the app.

### Priority

Medium. Anything that negatively affects the user experience.

### Description

The quantity of "Gold001" (0000000F) is not displayed correctly in the inventory. When receiving a reward in the form of coins, they are not displayed in the coin counter in the inventory. Also, when buying pies, the taken coins are not displayed in the coin counter in the inventory. At the same time, it is in the player's inventory, you can check it by buying or selling something.

### OS

Windows

### OS version

10

### Skyrim version

1.5,97,08

### SKSE version

2.0.20

### client commit''s hash

df63ebd83a2137064f4373d92fbee6a50f6ef75a

### server commit''s hash

df63ebd83a2137064f4373d92fbee6a50f6ef75a

### Videocard model

RX 470

### Steps to reproduce

1. Look at the amount of gold in your inventory
2. Buy a pie or sell an ingot
3. Open the inventory again

### Expected result

The amount of gold has changed

### Actual result

The amount of gold has not changed

### PR DESCRIPTION
**Pull request by Wedmer**
Added Game.ShowRacemenu, Game.ShowLimitedRacemenu, ObjectReference.GetPosition*, ObjectReference.SetPosition
This was made for SweetPie but should be useful for other gamemodes.

Removed unused script and added new for Mirror.
Updated Riften DM arena script.

Closes #953
Closes #997 as gold amount visualisation is vital for trading system

### CODE REVIEW

File: skymp5-server/cpp/server_guest_lib/PapyrusObjectReference.cpp [hunk 1]
```diff
@@ -273,3 +273,61 @@ VarValue PapyrusObjectReference::Activate(
...
+    if (arguments.size() < 3) {
+      throw std::runtime_error(
+        "SetPosition requires at least three arguments");
+    }
+    // TODO: Add movement/position changing sync for ingame objects and remove
+    // workaround below. Don't forget that SetPosition has fading effect. Also
+    // note that this change is global. So players should get new position on
+    // entering area or even outside of it
+    selfRefr->SetPosAndAngleSilent(
+      NiPoint3((float)static_cast<double>(arguments[0].CastToFloat()),
```
Comment (nic11) [inline skymp5-server/cpp/server_guest_lib/PapyrusObjectReference.cpp:317]: C-style cast after a C++ cast? Pleaseleave C++ one only...

File: skymp5-server/cpp/server_guest_lib/PapyrusObjectReference.cpp [hunk 2]
```diff
@@ -273,3 +273,61 @@ VarValue PapyrusObjectReference::Activate(
...
+               (float)static_cast<double>(arguments[1].CastToFloat()),
+               (float)static_cast<double>(arguments[2].CastToFloat())),
+      selfRefr->GetAngle());
+    selfRefr->ForceSubscriptionsUpdate();
+    auto funcName = "SetPosition";
+    auto s = SpSnippetFunctionGen::SerializeArguments(arguments);
```
Comment (nic11) [inline skymp5-server/cpp/server_guest_lib/PapyrusObjectReference.cpp:323]: Pls change thw name to something more understandable at a glance

File: unit/DistContentsExpected.json [hunk 3]
```diff
@@ -57,7 +57,9 @@
       "server/data/scripts/SweetEatAnim.pex",
       "server/data/scripts/SweetEatSoup.pex",
       "server/data/scripts/SweetKettleAnim.pex",
+      "server/data/scripts/SweetKettleAnim2.pex",
```
Comment (nic11) [inline unit/DistContentsExpected.json:60]: Did you forget the sources?

File: skymp5-server/cpp/server_guest_lib/PapyrusObjectReference.cpp (hunk 1 above)
Comment (Wedmer) [reply skymp5-server/cpp/server_guest_lib/PapyrusObjectReference.cpp:317]: That was copypaste from SetAngle

File: unit/DistContentsExpected.json (hunk 3 above)
Comment (Wedmer) [reply unit/DistContentsExpected.json:60]: They were already there

File: skymp5-server/cpp/server_guest_lib/PapyrusGame.cpp [hunk 4]
```diff
@@ -69,3 +69,30 @@ VarValue PapyrusGame::GetPlayer(VarValue self,
...
+
+VarValue PapyrusGame::ShowLimitedRaceMenu(
+  VarValue self, const std::vector<VarValue>& arguments)
+{
+  auto funcName = "ShowLimitedRaceMenu";
+  RaceMenuHelper(self, funcName, arguments);
+  return VarValue();
+}
+
+void PapyrusGame::RaceMenuHelper(VarValue& self, const char* funcName,
```
[the remark this answers is not in the document]
Comment (Wedmer) [reply skymp5-server/cpp/server_guest_lib/PapyrusGame.cpp:89]: RaceMenuImpl is at client side.

File: skymp5-server/cpp/server_guest_lib/PapyrusObjectReference.cpp [hunk 5]
```diff
@@ -314,17 +314,18 @@ VarValue PapyrusObjectReference::SetPosition(
     // note that this change is global. So players should get new position on
     // entering area or even outside of it
     selfRefr->SetPosAndAngleSilent(
-      NiPoint3((float)static_cast<double>(arguments[0].CastToFloat()),
-               (float)static_cast<double>(arguments[1].CastToFloat()),
-               (float)static_cast<double>(arguments[2].CastToFloat())),
+      NiPoint3(float(static_cast<double>(arguments[0].CastToFloat())),
```
Comment (nic11) [inline skymp5-server/cpp/server_guest_lib/PapyrusObjectReference.cpp:317]: This is still kinda C-style cast. Would it be ok to leave just `static_cast<double>` or `static_cast<float>` here? If double cast is needed here for some reason, then let's do a `static_cast` twice

**Comment by Pospelove** [comment]
I think nic11 is right about two `static_cast`s. Just for meeting our current guidelines.

**Comment by Pospelove** [comment]
I like this

File: skymp5-functions-lib/index.ts [hunk 6]
```diff
@@ -164,6 +164,164 @@ export const placeholder = (mp: Mp, self: PapyrusObject | null, args: PapyrusVal
...
+    reqs: [
+      {
+        formId: 0x0f,
+        requiredCount: 30,
+        returnSelect: 'random',
+        returns: [
+          {
+            formId: 0x0300353a,
+            returnNumber: 1,
+          }, {
```
Comment (nic11) [inline skymp5-functions-lib/index.ts:202]: Looks a bit strange, cuz all JSON pretty-printers (and you a few lines below) do like so
```suggestion
          },
          {
```

**Review by nic11 — APPROVED** [review]
Didn't read everything thoroughly (Papyrus code in particular), but LGTM

File: skymp5-functions-lib/index.ts (hunk 6 above)
Comment (Wedmer) [reply skymp5-functions-lib/index.ts:202]: Copypaste went wrong

**Pull request merged**